src/ui/analysis.jsx

```diff
@@ -0,0 +1,125 @@
+import React from 'react';
+import moment from 'moment';
+import DataStoreContext from './datastore';
+
+/**
+ * A singleton class for calculating analysis of a run.
+ *
+ * This is full of `get` methods, which use React hooks to ensure that long calculations are
+ * re-done only when necessary.  All of these functions should only be called within a
+ * render function.
```

Comment:
What is a render function in react hooks lingo?

Reply:
It's the same as non-hooks lingo -- either `render()` on a component class or, for a pure functional component, the function itself.

```javascript
const MyCompoent = (props) => {
  // call hooks stuff in here
};
```

Reply:
From https://reactjs.org/docs/hooks-faq.html#should-i-use-hooks-classes-or-a-mix-of-both, it looks like class components are not encouraged so render() methods won't be required anymore.

Reply:
That sounds good to me!  Hooks are actually quite nice once you get used to them.  Is "render function" the right terminology, then?

Reply:
It's basically just the return value of the function component. I'm not sure if I would call it render function but I guess that's better.